grpc-client-server/prober_server/main.go

```diff
@@ -23,13 +33,35 @@ type server struct {
 }
 
 func (s *server) DoProbes(ctx context.Context, in *pb.ProbeRequest) (*pb.ProbeReply, error) {
-	// TODO: support a number of repetitions and return average latency
-	start := time.Now()
-	_, _ = http.Get(in.GetEndpoint())	// TODO: add error handling here and check the response code
-	elapsed := time.Since(start)
-	elapsedMsecs := float32(elapsed / time.Millisecond)
-
-	return &pb.ProbeReply{LatencyMsecs: elapsedMsecs}, nil
+	var sumOfelapsedMsecs = float32(0)
+	numberOfRepeats := in.GetNumberOfRequestsToMake()
+	var replies = make([]*pb.Reply, 0)
```

Comment:
It's not important (it's a micro-optimisation), but when we know how many elements we're (likely) going to add to a slice, we can pre-allocate capacity for it:
```suggestion
	var replies = make([]*pb.Reply, 0, numberOfRepeats)
```

I don't know what the default capacity of a slice is, but let's imagine it's either 1 or 128 elements - what advantage does setting a capacity have over both possible defaults?

Reply:
Ah, I did forget that there is a default capacity. Good point!